VMBackup/main/freezesnapshotter.py

```diff
@@ -170,7 +175,27 @@ def update_snapshotinfoarray(self, blob_snapshot_info_array):
         if blob_snapshot_info_array != None and blob_snapshot_info_array !=[]:
             for blob_snapshot_info in blob_snapshot_info_array:
                 if blob_snapshot_info != None:
-                    snapshot_info_array.append(Status.SnapshotInfoObj(blob_snapshot_info.isSuccessful, blob_snapshot_info.snapshotUri, blob_snapshot_info.errorMessage))
+                    self.logger.log("IsSuccessful:{0}, SnapshotUri:{1}, ErrorMessage:{2}".format(blob_snapshot_info.isSuccessful, blob_snapshot_info.snapshotUri, blob_snapshot_info.errorMessage))
+
+                    blobUri = blob_snapshot_info.snapshotUri
+                    if(not blob_snapshot_info.snapshotUri):
+                        endIndexOfBlobUri = blob_snapshot_info.snapshotUri.find('?')
+                        if(blob_snapshot_info.DDSnapshotIdentifier != None):
+                            endIndexOfBlobUri = blob_snapshot_info.snapshotUri.find("/snapshots")
+                        if(endIndexOfBlobUri != -1):
+                            blobUri = blobUri[0:endIndexOfBlobUri]
+                        else:
+                            self.logger.log("Unable to find end index of blobUri in snapshotUri. Assigning default snapshotUri to blobUri. This {0} a DirectDrive disk".format("is" if(blob_snapshot_info.DDSnapshotIdentifier != None) else "is not"))
+                    self.logger.log("blobUri : {0}".format(blobUri))
+                        
+                    ddSnapshotIdentifierInfo = blob_snapshot_info.DDSnapshotIdentifier
+                    if(ddSnapshotIdentifierInfo != None):
+                        ddSnapshotIdentifierInfo = Status.DirectDriveSnapshotIdentifier(ddSnapshotIdentifierInfo.creationTime, ddSnapshotIdentifierInfo.id, ddSnapshotIdentifierInfo.token)
```

Comment:
minor : as DDSnapshotIdentifier and DirectDriveSnapshotIdentifier are two different classes, to avoid confusion, we can initialize DirectDriveSnapshotIdentifier with None 

                    directDriveSnapshotIdentifier = None
                    if (blob_snapshot_info.DDSnapshotIdentifier != None):